src/wovnio/wovnphp/Headers.php

```diff
@@ -313,19 +313,44 @@ public function removeLang($uri, $lang = null)
 
         $lang_code = $this->store->convertToCustomLangCode($lang);
         $default_lang = $this->store->settings['default_lang'];
-        if ($this->store->hasDefaultLangAlias()) {
+        if ($lang_code && $this->store->hasDefaultLangAlias()) {
             $no_lang_uri = Url::removeLangCode($uri, $lang_code, $this->store, $this);
             return Url::addLangCode($no_lang_uri, $this->store, $default_lang, $this);
         } else {
             return Url::removeLangCode($uri, $lang_code, $this->store, $this);
         }
     }
 
+    public function canProcessResponse()
+    {
+        # To process a response means to add snippet/hreflangs and translate
+
+        // query pattern doesn't affect which source page is requested
+        // custom domain pattern isn't affected by lang aliases
+        $defaultLangAliasHasLimitedScope = $this->store->settings['url_pattern_name'] == 'path'
+                                        || $this->store->settings['url_pattern_name'] == 'subdomain';
+        if ($defaultLangAliasHasLimitedScope) {
+            $urlLanguage = $this->urlLanguage();
+            $urlLanguageIsEmpty = !$urlLanguage || strlen($urlLanguage)==0;
```

Comment:
Do you know which case we need `strlen($urlLanguage)==0` ?

Reply:
Reference
https://leben.mobi/blog/php_isset_etc/php/

Reply:
I just copied this check from somewhere. I agree it isn't useful but not sure why it is there.

Reply:
https://github.com/WOVNio/wovnphp/blame/master/src/wovnio/wovnphp/Headers.php#L148 I don't think we'll know because it was from the first commit of the old wovnphp